integreat_cms/cms/views/utils/hix.py

```diff
@@ -59,13 +60,21 @@ def get_hix_score(request, region_slug):
     # Don't pass texts larger than 100kb to the api in order to avoid being vulnerable to dos attacks
     if len(request.body) > MAX_TEXT_LENGTH:
         return JsonResponse({"error": "Request too large"})
+
     body = json.loads(request.body.decode("utf-8"))
     text = body["text"]
 
     if not isinstance(text, str) or not text.strip():
         logger.warning("Received invalid text: %r", text)
         return JsonResponse({"error": f"Invalid text: '{text}'"})
 
-    if score := lookup_hix_score(text):
+    # convert special characters into plain letters
+    unescaped = html.unescape(text.replace("\n", "\r\n"))
+    # convert special characters to numeric reference
+    to_numeric_reference = unescaped.encode("ascii", "xmlcharrefreplace")
+    # convert from byte to string
+    to_string = to_numeric_reference.decode("utf-8", "xmlcharrefreplace")
```

Comment:
> Sadly I couldn't find a way to convert entity reference to numeric reference directly and needed some converting steps.

I think the different encoding is a result from our [custom clean method](https://github.com/digitalfabrik/integreat-cms/blob/0dc0592a301cf233300844a0a65e204e87f947da/integreat_cms/cms/forms/custom_content_model_form.py#L103), so you could also do:
```
from lxml.etree import LxmlError
from lxml.html import fromstring, tostring

try:
    text = tostring(fromstring(text)).decode("utf-8")
except LxmlError:
    pass
```
but that also isn't that much shorter...

Reply:
Applied 🙂